wordSearch.py

```diff
@@ -0,0 +1,35 @@
+# As taught in class, I used backtracking to find the required word
+# Time Complexity: exponeential
+#Space Complexity: O(1) as I manipulate the saame array
```

Comment:
Time: O((n * m) * 3^l), one recursive call can be made for each space in the 2-D matrix, l is the length of the word.
Space: O(n * m), recursive stack space.